velox/connectors/hive/storage_adapters/hdfs/tests/HdfsInsertTest.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "gtest/gtest.h"
+
+#include "velox/common/memory/Memory.h"
+#include "velox/connectors/hive/storage_adapters/hdfs/RegisterHdfsFileSystem.h"
+#include "velox/connectors/hive/storage_adapters/hdfs/tests/HdfsMiniCluster.h"
+#include "velox/connectors/hive/storage_adapters/test_common/InsertTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox::connector;
+using namespace facebook::velox::test;
+
+class HdfsInsertTest : public testing::Test, public InsertTest {
+ public:
+  static void SetUpTestSuite() {
+    memory::MemoryManagerOptions options;
+    memory::MemoryManager::testingSetInstance(options);
+  }
+
+  void SetUp() override {
+    connector::registerConnectorFactory(
+        std::make_shared<connector::hive::HiveConnectorFactory>());
+    filesystems::registerHdfsFileSystem();
+    if (miniCluster == nullptr) {
+      miniCluster = std::make_shared<filesystems::test::HdfsMiniCluster>();
+      miniCluster->start();
+    }
+    ioExecutor_ = std::make_unique<folly::IOThreadPoolExecutor>(3);
+
+    auto hiveConnector =
+        connector::getConnectorFactory(
+            connector::hive::HiveConnectorFactory::kHiveConnectorName)
+            ->newConnector(
+                exec::test::kHiveConnectorId,
+                std::make_shared<const config::ConfigBase>(
+                    std::unordered_map<std::string, std::string>()),
+                ioExecutor_.get());
+    connector::registerConnector(hiveConnector);
+    parquet::registerParquetReaderFactory();
+    parquet::registerParquetWriterFactory();
```

Comment:
It looks like we are bringing more code into the SetUp process. Does it make sense to also extends `HiveConnectorTestBase` to minimize the change? Thanks.

Reply:
HiveConnectorTestBase should not be linked with Parquet or other components.

Reply:
We could have a ParquetTestBase.

Reply:
@rui-mo Yes, additional code has been introduced to register and unregister the Parquet reader and writer factory in the `GcsInsertTest`, `S3InsertTest`, and `HdfsInsertTest`. And we plan to submit a separate PR to relocate this code to a common location. As suggested by @majetideepak , we could also integrate this code into the `ParquetTestBase`.